src/lib.rs

```diff
@@ -0,0 +1,270 @@
+use log::trace;
+use std::{collections, fs, path};
+
+pub struct OverdropConf {
+    dirs: Vec<path::PathBuf>,
+}
+
+impl OverdropConf {
+    pub fn new(
+        base_dirs: &Vec<String>,
+        config_path: &str,
+    ) -> Self {
+        let mut dirs = Vec::with_capacity(base_dirs.len());
+        for bdir in base_dirs {
+            let mut dpath = path::PathBuf::from(bdir);
+            dpath.push(config_path.clone());
+            dirs.push(dpath);
+        }
+        Self { dirs }
+    }
+
+    pub fn new_full(
+        root_dir: &str,
+        base_dirs: &Vec<String>,
+        config_path: &str,
+        version: Option<String>,
+    ) -> Self {
+        let mut dirs = Vec::with_capacity(base_dirs.len());
+        let version = version.unwrap_or("".to_string());
+        for bdir in base_dirs {
+            let mut dpath = path::PathBuf::from(root_dir);
```

Comment:
Do we think we need this, outside of tests? If not, I'd leave this prefixing to consumers and just keep a single `new()` implementation.

Reply:
The idea for `root_dir` this only came up while writing a test - agreed on leaving the prefixing to consumers.